device-types/Datacom/DM4770-32CX.yaml

```diff
@@ -0,0 +1,129 @@
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/8
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/9
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/10
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/11
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/12
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/13
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/14
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/15
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/16
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/17
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/18
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/19
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/20
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/21
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/22
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/23
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/24
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/25
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/26
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/27
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/28
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/29
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/30
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/31
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: hundred-gigabit-ethernet 1/1/32
+    type: 100gbase-x-qsfp28
+    description: 100G/40G Base-X (QSFP28)
+  - name: mgmt-1/1/1
+    type: 100base-tx
```

Comment:
If this is a OOB management port, you should mark it as such with `mgmt_only: true`